From 25641e0fbe901c2826a0e74a45cbb46313f8ec81 Mon Sep 17 00:00:00 2001
From: "N, Shyjumon" <shyjumon.n@intel.com>
Date: Sat, 22 Aug 2020 00:01:43 +0530
Subject: [PATCH 4/4] Enable wifi HAL for Android R

TODO: hostapd changes need to be revisited

Tracked-On: None
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Signed-off-by: N, Shyjumon <shyjumon.n@intel.com>
---
 .../caas/framework_manifest.xml               | 27 +++++++++++++++++++
 groups/device-specific/caas/manifest.xml      |  8 +++---
 2 files changed, 31 insertions(+), 4 deletions(-)

diff --git a/groups/device-specific/caas/framework_manifest.xml b/groups/device-specific/caas/framework_manifest.xml
index ff06646..59e5cbd 100644
--- a/groups/device-specific/caas/framework_manifest.xml
+++ b/groups/device-specific/caas/framework_manifest.xml
@@ -141,6 +141,33 @@
             <name>IBluetoothHci</name>
             <instance>default</instance>
         </interface>
+    </hal>
+	<hal format="hidl">
+        <name>android.hardware.wifi.supplicant</name>
+        <transport>hwbinder</transport>
+        <version>1.3</version>
+        <interface>
+            <name>ISupplicant</name>
+            <instance>default</instance>
+        </interface>
+    </hal>
+    <hal format="hidl">
+        <name>android.hardware.wifi</name>
+        <transport>hwbinder</transport>
+        <version>1.4</version>
+        <interface>
+            <name>IWifi</name>
+            <instance>default</instance>
+        </interface>
+    </hal>
+    <hal format="hidl">
+        <name>android.hardware.wifi.hostapd</name>
+        <transport>hwbinder</transport>
+        <version>1.2</version>
+        <interface>
+            <name>IHostapd</name>
+            <instance>default</instance>
+        </interface>
     </hal>
     <hal format="hidl">
         <name>android.hardware.media.omx</name>
diff --git a/groups/device-specific/caas/manifest.xml b/groups/device-specific/caas/manifest.xml
index ea722e3..85be8f8 100644
--- a/groups/device-specific/caas/manifest.xml
+++ b/groups/device-specific/caas/manifest.xml
@@ -145,7 +145,7 @@
     <hal format="hidl">
         <name>android.hardware.wifi.supplicant</name>
         <transport>hwbinder</transport>
-        <version>1.1</version>
+        <version>1.3</version>
         <interface>
             <name>ISupplicant</name>
             <instance>default</instance>
@@ -154,7 +154,7 @@
     <hal format="hidl">
         <name>android.hardware.wifi</name>
         <transport>hwbinder</transport>
-        <version>1.3</version>
+        <version>1.4</version>
         <interface>
             <name>IWifi</name>
             <instance>default</instance>
@@ -163,7 +163,7 @@
     <hal format="hidl">
         <name>android.hardware.wifi.hostapd</name>
         <transport>hwbinder</transport>
-        <version>1.1</version>
+        <version>1.2</version>
         <interface>
             <name>IHostapd</name>
             <instance>default</instance>
@@ -294,6 +294,6 @@
     </hal>

     <sepolicy>
-        <version>27.0</version>
+        <version>30.0</version>
     </sepolicy>
 </manifest>
--
2.17.1

